cpu/sam0_common/periph/flashpage.c

```diff
@@ -45,19 +42,67 @@ void flashpage_write(int page, void *data)
         PAC1->WPCLR.reg = NVMCTRL_PAC_BIT;
     }
 #endif
+}
+
+static void _lock(void)
+{
+    /* put peripheral access lock for the NVMCTRL peripheral */
+#ifdef CPU_FAM_SAML21
+    PAC->WRCTRL.reg = (PAC_WRCTRL_KEY_SET | ID_NVMCTRL);
+#else
+    if (PAC1->WPCLR.reg & NVMCTRL_PAC_BIT) {
+        PAC1->WPSET.reg = NVMCTRL_PAC_BIT;
+    }
+#endif
+}
+
+void flashpage_write_raw(void *target_addr, void *data, size_t len)
+{
+    /* The actual minimal block size for writing is 16B, thus we
+     * assert we write on multiples and no less of that length.
+     */
+    assert(!(len % FLASHPAGE_RAW_BLOCKSIZE));
+
+    /* ensure 4 byte aligned writes */
+    assert(!(((unsigned)target_addr % FLASHPAGE_RAW_ALIGNMENT) ||
+            ((unsigned)data % FLASHPAGE_RAW_ALIGNMENT)));
+
+    /* ensure the length doesn't exceed the actual flash size */
+    assert((target_addr + len) < (CPU_FLASH_BASE + (FLASHPAGE_SIZE * FLASHPAGE_NUMOF)));
```

Comment:
Pointer-arithmetics on void pointer. Please cast `target_addr` to `uint8_t *`. See also [static test results](https://ci.riot-os.org/RIOT-OS/RIOT/7970/4cfbb1ecb651e604966da0fdabe889e8fdfa1ead/output/static_tests.txt)

Reply:
> Please cast `target_addr` to `uint8_t *`

Hmm... it wouldn't be more correct `uint32_t*`?

Reply:
Depends on what you are trying to do here `((uint8_t *)target_addr + len)` is the address of the byte `len` bytes after `target_addr`, `((uint32_t *)target_address + len)` is the address of the byte `len` *\* 4* after `target_addr` (because `sizeof(uint32_t) == sizeof(uint8_t) * 4`).

Reply:
Addressed.